pelican/indexd.py

```diff
@@ -0,0 +1,37 @@
+import requests
+import json
+import base64
+
+def indexd_submit(hostname, access_token, file_name, size, urls, hashes, did = None):
+	"""
+	:params hostname: url of the commons
+	"params access_token: token for submitting files to indexd"
+	:params file_name: path to file
+	:params size: size of the file
+	:parmas urls: array of urls of the file (S3 urls)
+	:params hashes: dictionary of file hashes {s3: 1234}
+	:params did: UUID of the file, Optional as indexd will create a did if not specified
+	"""
+
+	body = {}
+	body["file_name"] = file_name
+	body["size"] = size
+	body["urls"] = urls
+	body["hashes"] = hashes
+	body["form"] = "object"
+	if did:
+		body["did"] = did
+
+	indexd_hostname = hostname + "index/index"
+
+	print("-----------------------------------------------------")
+	print(indexd_hostname)
+	print(json.dumps(body))
+	print("-----------------------------------------------------")
```

Comment:
i think there's an important step missing: the ACL calculation for the indexd record using cohort records. if it's a lot of work u can make a new ticket. see design doc: https://docs.google.com/document/d/1954SSzVgDDM41uBksIt9RY7_MRAcCzoaMfmwHtfpFG4/edit#

"An authz list is constructed that combines, in an AND operation, all program/project paths from the set of auth_resource_paths for the records in the cohort." i think you'd need to insert new records into the index_record_authz table